### PR TITLE
Adds FF library plugin

### PR DESCRIPTION
**Pull request by Steve-Mcl**
## Description

NOTE: This is part 2 of 2 PRs to enable the feature described in https://github.com/flowforge/flowforge/issues/2294

1. Adds duplicate of `flowforge-nr-launcher/lib/storage/libraryPlugin.js`
2. Checks and stores new boolean property `licensed` to include (or exclude) the necessary settings for loading the plugin
3. Plugin integration is made via `nodesDir` and points to the plugin path within the agent
   1. NOTE: a dummy package.json file was also added alongside the plugin since it would not load without it
4. adds test `updates licensed state`
5. updates existing tests to ensure correct presence and state of the `licensed` flag

## Related Issue(s)

https://github.com/flowforge/flowforge/issues/2294

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `flowforge/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `flowforge/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,6 +2,14 @@
     "name": "@flowforge/flowforge-device-agent",
     "version": "1.9.4",
     "description": "An Edge Agent for running Node-RED instances deployed from the FlowForge Platform",
+    "exports": {
+        "./libraryPlugin": "./lib/plugins/libraryPlugin.js"
+    },
+    "node-red": {
+        "plugins": {
+            "flowforge-library": "lib/plugins/libraryPlugin.js"
```
Comment (Steve-Mcl) [inline package.json:10]:
```suggestion
        "./libraryPlugin": "./lib/plugins/node_modules/libraryPlugin.js"
    },
    "node-red": {
        "plugins": {
            "flowforge-library": "lib/plugins/node_modules/libraryPlugin.js"
```

Comment (knolleary) [reply package.json:10]: I don't think the device agent's package.json should have a node-red section at all - it isn't a module that is intended to provide plugins to anything other than the NR instance it spawns.

Comment (Steve-Mcl) [reply package.json:10]: Agreed - we can add later if that becomes necessary. Will remove

File: package.json [hunk 2]
```diff
@@ -2,6 +2,14 @@
...
+        }
+    },
```
Comment (Steve-Mcl) [inline package.json:12]:
```suggestion
```

Remove "node-red" section - not required right now.

**Review by knolleary — APPROVED** [review]
Looks good. Thanks @Steve-Mcl

**Pull request merged**